src/views/TimeOutWrapper/index.tsx

```diff
@@ -12,27 +17,163 @@ const sessionTimeout = { minutes: 14 };
 
 const registerExpire = async (authService: any, lastActiveAt: number) => {
   const tokenManager = await authService.getTokenManager();
+
   // clear the old listener so we don't register millions of them
   tokenManager.off('expired');
   tokenManager.on('expired', (key: any) => {
     const activeSessionWindow = DateTime.local()
       .minus(sessionTimeout)
       .toMillis();
-    if (lastActiveAt > activeSessionWindow && key === 'accessToken') {
+    if (
+      lastActiveAt > activeSessionWindow &&
+      ['idToken', 'accessToken'].includes(key)
+    ) {
       tokenManager.renew(key);
     }
   });
 };
 
+/**
+ * As of okta-react 3.0.1, autoRenew on the Security component does
+ * not work. We need to manually renew the session and access token.
+ */
 const TimeOutWrapper = ({ children }: TimeOutWrapperProps) => {
+  const dispatch = useDispatch();
+  const { authState, authService } = useOktaAuth();
+  const { t } = useTranslation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const sessionInterval = useRef<any>();
+  const [timeRemainingArr, setTimeRemainingArr] = useState([0, 'second']);
   const lastActiveAt = useSelector(
     (state: AppState) => state.auth.lastActiveAt
   );
 
-  const { authService }: { authService: any } = useOktaAuth();
-  registerExpire(authService, lastActiveAt);
+  const sessionExpiration = useSelector(
+    (state: AppState) => state.auth.sessionExpiration
+  );
+
+  const oneSecond = Duration.fromObject({ seconds: 1 }).as('milliseconds');
+  const oneMinute = Duration.fromObject({ minutes: 1 }).as('milliseconds');
+  const fiveMinutes = Duration.fromObject({ minutes: 5 }).as('milliseconds');
+
+  const storeSession = async () => {
+    // eslint-disable-next-line no-underscore-dangle
+    const session = await authService._oktaAuth.session.get();
+    const expiresAt = new Date(session.expiresAt).getTime();
+    dispatch(updateLastSessionRenew(expiresAt));
```

Comment:
Why store this in the reducer instead of just relying on the value we fetched from the token? I worry that the two might get out of step, and we want the token to be the source of truth.